src/ruby/spec/pb/package_with_underscore/checker_spec.rb

```diff
@@ -0,0 +1,50 @@
+# Copyright 2016 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require 'open3'
+require 'tmpdir'
+
+describe 'Package with underscore protobuf code generation' do
+  it 'should have the same content as created by code generation' do
+    root_dir = File.join(File.dirname(__FILE__), '..', '..', '..', '..', '..')
+    pb_dir = File.join(root_dir, 'src', 'ruby', 'spec', 'pb')
+
+    bins_sub_dir = ENV['DEBUG'].nil? ? 'opt' : 'dbg'
```

Comment:
it looks like you're checking the wrong environment variable here. Change `DEBUG` to `CONFIG` and this should work (see https://github.com/grpc/grpc/blob/master/tools/run_tests/run_tests.py#L142)

FYI one way to run this test both configs locally is by running

`tools/run_tests/run_tests.py -l ruby -c dbg`

and 

`tools/run_tests/run_tests.py -l ruby -c opt`